pyccel/parser/semantic.py

```diff
@@ -4176,22 +4215,61 @@ def _visit_FunctionalFor(self, expr):
             errors.report(LIST_OF_TUPLES, symbol=expr, severity='error')
 
         target.invalidate_node()
+        operations = []
+        assign = None
+        target_conversion_func = self._visit(target_type_name)
+        if (isinstance(target_conversion_func, PyccelFunctionDef)
+            and target_conversion_func.cls_name is NumpyArray):
```

Comment:
For readability so the condition does not look like the start of the code
```suggestion
        if (isinstance(target_conversion_func, PyccelFunctionDef)
                and target_conversion_func.cls_name is NumpyArray):
```